### PR TITLE
KAFKA-3010; Include error in log when ack 0

### PR DESCRIPTION
**Pull request by ijuma**
I verified this by trying to produce to __consumer_offsets and the logged message looks like:

[2015-12-22 10:34:40,897] INFO [KafkaApi-0] Closing connection due to error during produce request with correlation id 1 from client id console-producer with ack=0
Topic and partition to exceptions: [__consumer_offsets,43] -> kafka.common.InvalidTopicException (kafka.server.KafkaApis)

### CODE REVIEW

File: core/src/main/scala/kafka/server/KafkaApis.scala [hunk 1]
```diff
@@ -315,19 +315,23 @@ class KafkaApis(val requestChannel: RequestChannel,
...
-        // we only print warnings for known errors here; if it is unknown, it will cause
-        // an error message in the replica manager
-        if (status.error != ErrorMapping.NoError && status.error != ErrorMapping.UnknownCode) {
-          debug("Produce request with correlation id %d from client %s on partition %s failed due to %s".format(
-            produceRequest.correlationId,
-            produceRequest.clientId,
-            topicAndPartition,
-            ErrorMapping.exceptionNameFor(status.error)))
+        if (status.error != ErrorMapping.NoError) {
           errorInResponse = true
```
Comment (ijuma) [inline core/src/main/scala/kafka/server/KafkaApis.scala:322]: This change means that we will log the "Closing connection" message even if the error is unknown. It seems to me that it's what we want, but I'd like someone more familiar with this code to confirm. cc @junrao

Comment (gwenshap) [inline core/src/main/scala/kafka/server/KafkaApis.scala:322]: We want to log the message even if error is unknown, but perhaps log the error code in that case? I'm not sure this should happen on the broker-side, but will be nice to have some information if it does.

Comment (ijuma) [inline core/src/main/scala/kafka/server/KafkaApis.scala:322]: We log the exception name which is UnknownException, which I think is more friendly than -1 (the error code). What do you think?

File: core/src/main/scala/kafka/server/KafkaApis.scala [hunk 2]
```diff
@@ -315,19 +315,23 @@ class KafkaApis(val requestChannel: RequestChannel,
...
+          // we only print warnings for known errors here; if it is unknown, it will cause
+          // an error message in the replica manager
+          if (status.error != ErrorMapping.UnknownCode) {
```
Comment (junrao) [inline core/src/main/scala/kafka/server/KafkaApis.scala]: Perhaps we should just remove the if test and log UnknownException as well.

Comment (ijuma) [inline core/src/main/scala/kafka/server/KafkaApis.scala]: Sounds good. Many other requests have similarly looking code, should those be changed as well or should we only change it here?

Comment (junrao) [inline core/src/main/scala/kafka/server/KafkaApis.scala]: The two other usages are in debug logging too. It won't hurt to log the same thing in debug mode again. So, we can log UnknownExecption in those cases too.

Comment (ijuma) [inline core/src/main/scala/kafka/server/KafkaApis.scala]: Will do.

**Comment by ijuma** [comment]
@junrao I made the change you requested. Tests pass locally.

**Comment by junrao** [comment]
Thanks for the patch. LGTM

**Pull request closed without merging**